packages/react-storage/src/components/StorageBrowser/actions/configs/types.ts

```diff
@@ -80,63 +84,93 @@ export interface TaskActionConfig<T extends TaskHandler> {
   /**
    * whether the action allows inflight cancellation
    */
-  isCancelable: IsCancelableTaskHandler<T>;
+  isCancelable: boolean;
 
   /**
    * action handler
    */
   handler: T;
 
   /**
-   * default title value displayed on action view
+   * default display name value displayed on action view
    */
-  title: string;
+  displayName: string;
+}
 
+export interface BatchTaskActionConfig<
+  T extends TaskHandler,
+  ComponentName extends Capitalize<string> = Capitalize<string>,
+> extends TaskActionConfig<T, ComponentName> {
   /**
    * sets action view subcomponent type and action processing behavior
    */
-  type: 'BATCH_ACTION' | 'SINGLE_ACTION';
+  type: 'BATCH_ACTION';
 }
 
-export interface ListActionConfig<T> extends ActionConfigTemplate<T> {}
+export interface SingleTaskActionConfig<
+  T extends TaskHandler,
+  ComponentName extends Capitalize<string> = Capitalize<string>,
+> extends TaskActionConfig<T, ComponentName> {
+  /**
+   * sets action view subcomponent type and action processing behavior
+   */
+  type: 'SINGLE_ACTION';
+}
 
-export interface UploadActionConfig extends TaskActionConfig<UploadHandler> {}
+export interface ListActionConfig<T, ComponentName>
+  extends ActionConfigTemplate<T> {
+  componentName: ComponentName;
+}
+
+export interface UploadActionConfig
+  extends TaskActionConfig<UploadHandler, 'UploadView'> {
+  type: 'BATCH_ACTION';
+  isCancelable: IsCancelableTaskHandler<UploadHandler>;
+}
 
 export interface CreateFolderActionConfig
-  extends TaskActionConfig<CreateFolderHandler> {}
+  extends TaskActionConfig<CreateFolderHandler, 'CreateFolderView'> {
+  type: 'SINGLE_ACTION';
+  isCancelable: IsCancelableTaskHandler<CreateFolderHandler>;
+}
 
 export interface ListLocationsActionConfig
-  extends ListActionConfig<ListLocationsHandler> {
-  title: string;
+  extends ListActionConfig<ListLocationsHandler, 'ListLocationsView'> {
+  displayName: string;
   type: 'LIST_LOCATIONS';
```

Comment:
Is it possible to remove `type` for the `LocationsView`/`LocationDetailsView` default configs and only associate `type` with batch/single action configs?

Reply:
I thought `LocationsView` and `LocationDetailsView` have different sets of the "sub components" from each other, the type is used to determine the difference [here](https://github.com/aws-amplify/amplify-ui/pull/5907/files#diff-de7a77d9802b8dfe03582f16330828f5d6dcb60c8a8c925a3935fd78cf7b2b57R102-R110). Is the difference on the "sub components" still exists? if not we can surely remove the type.